File: baekjoon/BinarySearch/10815_숫자 카드.py
import sys
input = sys.stdin.readline

n = int(input())
nArr = list(map(int, input().split(" ")))
nArr.sort()
m = int(input())
mArr = list(map(int, input().split(" ")))


def solution():
    answer = []
    for find in mArr:
        start = 0
        end = n-1
        check = False
        while start <= end:
            mid = (start+end)//2
            if nArr[mid] == find:
                check = True
                break
            elif nArr[mid] > find:
                end = mid - 1
            else:
                start = mid + 1
        answer.append(1 if check else 0)

    return " ".join(map(str, answer))


print(solution())

File: baekjoon/BinarySearch/10815_숫자 카드.js
const fs = require('fs')
let [n, nArr, m, mArr] = fs
  .readFileSync('../input.txt')
  .toString()
  .trim()
  .split('\n')
n = Number(n)
nArr = nArr
  .split(' ')
  .map(Number)
  .sort((a, b) => a - b)
mArr = mArr.split(' ').map(Number)

function solution() {
  const answer = []
  for (const find of mArr) {
    let start = 0
    let end = n - 1
    let check = false

    while (start <= end) {
      let mid = parseInt((start + end) / 2)
      if (nArr[mid] === find) {
        check = true
        break
      } else if (nArr[mid] > find) {
        end = mid - 1
      } else {
        start = mid + 1
      }
    }
    answer.push(check ? 1 : 0)
  }
  return answer.join(' ')
}

console.log(solution())
